packages/babel-core/src/transformation/file/options/option-manager.js

```diff
@@ -174,6 +174,10 @@ export default class OptionManager {
     loc = loc || alias;
 
     for (let key in opts) {
+      // Ignore Babel's ES2015 module sentinel on the `opts` object, if any. This
+      // is so Babel presets may be written using ES2015 modules syntax.
+      if (key === "__esModule") continue;
```

Comment:
Per my other comment, it feels like rather than doing this here, we could do it here: https://github.com/johanssj/babel/blob/5fe291616d8bc99f6048cdcabe8fa828c2e68fd4/packages/babel-core/src/transformation/file/options/option-manager.js#L274 e.g.

    if (val.__esModule) val = val.default;